### DISCUSSION TITLE
Excluding assets from _site build folder

### DISCUSSION THREAD

**Discussion by Jip-Hop** [mmistakes/minimal-mistakes#2729] [Q&A]
Maybe I misunderstand how the `exclude` option in `_config.yml` is supposed to work.
But by default `assets/js/plugins` is among the excluded folders:

https://github.com/mmistakes/minimal-mistakes/blob/master/_config.yml#L156

Yet the folder and its contents end up in the `_site` folder when building and are published to GitHub Pages:
https://mmistakes.github.io/minimal-mistakes/assets/js/plugins/gumshoe.js

Since these are included in the minified JS, I don't see a reason why they need to be included in the `_site` folder. How can I exclude them?

On a related note I was trying to exclude `_site/assets/css/main.css` from being generated since I inlined my style in the head. But excluding it in`_config.yml` isn't helping.

P.S.
Great theme!

**Comment by mmistakes** [comment]
Not sure. How you're expecting `exclude` to work is how I thought it worked as well. Maybe it's a bug with Jekyll?

I seem to remember some issues several releases ago where they added globbing support for the `exclude` and `include` options and it had some unintended side affects. Maybe this is related.

**Comment by Jip-Hop** [comment]
Glad to hear it's not just me. I came across https://github.com/jekyll/jekyll/issues/5909. Perhaps it's caused by the `github-pages` gem?

**Comment by akhyarrh** [comment]
Using Jekyll 4 still yield same issue for me. If you are not using Github to build Jekyll, you can use [this action](https://github.com/peaceiris/actions-gh-pages) and use `exclude_assets` option to not include unwanted file.
